account_reconcile_oca/models/account_bank_statement_line.py

```diff
@@ -244,21 +245,21 @@ def _recompute_suspense_line(self, data, reconcile_auxiliary_id, manual_referenc
             if line["kind"] != "suspense":
                 new_data.append(line)
                 total_amount += line["amount"]
-                if line.get("currency_amount"):
-                    currency_amount += (
-                        self.env["res.currency"]
-                        .browse(line["line_currency_id"])
-                        ._convert(
-                            line["currency_amount"],
-                            self._get_reconcile_currency(),
-                            self.company_id,
-                            self.date,
-                        )
-                    )
+                # case of statement line with foreign_currency
+                if (
+                    line["kind"] == "liquidity"
+                    and line["line_currency_id"] != suspense_currency.id
+                ):
+                    currency_amount += self.amount_currency
+                elif (
+                    line.get("currency_amount")
+                    and line.get("line_currency_id") == suspense_currency.id
+                ):
+                    currency_amount += line.get("currency_amount")
```

Comment:
Avoid conversion if we already have an amount_currency in the same currency of our suspense line